R/add_custom_effect_sizes.R

```diff
@@ -0,0 +1,39 @@
+
+
+add_custom_effect_sizes <- function(estimate, ...) {
+  UseMethod("add_custom_effect_sizes", estimate)
+}
+
+
+add_custom_effect_sizes.character <- function(estimate, canonical_table, .x = NULL, ...) {
+
+  if(length(estimate) > 1L) warning("Calculating more than one effect-size measure is now deprecated.")
+  estimate <- estimate[[1L]]
+
+  add_effect_sizes(x = canonical_table, es = estimate, ...)
+}
+
+
+add_custom_effect_sizes.data.frame <- function(estimate, canonical_table, intercept = FALSE, ...) {
+
+  if(!intercept) canonical_table <- canonical_table[canonical_table$term != "(Intercept)", , drop = FALSE]
+
+  y <- merge(canonical_table,
+             tidy_es(estimate)
+             , sort = FALSE
+             , all.x = TRUE # Do not drop
+  )
+  if(anyNA(y)) {
+    warning("Custom effect sizes were not available for some model terms. These have been dropped from the output object.", call. = FALSE)
```

Comment:
Hmm, when does this typically happen? And wouldn't it make more sense to retain the term but just omit the effect size?

Reply:
If the user chooses an effect-size function that does not calculate effect sized for intercept terms, this will happen. I think it is better to warn here, because this will most likely be a mistake. But I will remove the dropping-terms part, maybe that's a bit harsh.

Reply:
If I leave the NA, there are some problems that need to be addressed downstream: If we allow NAs in an `apa_results_table`, how should they be included? Should it be `"NA"` or `NA_character_`? I'm in favor of `NA_character_`, but this will need to be addressed within `beautify` and possibly `glue_apa_results()`.

Reply:
`printnum()` has an argument, which draws on a global option to process this, no? But if it's too much of a hassle, just keep it as is!

Reply:
OK, after playing around with this one I determined it is best actually the best solution to drop the terms with NAs. Keeping terms with NAs, I had to switch off lots of input validation that is happening down the road, and we would have to write lots of code handling these rare (and most likely non-intentional) cases.